airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -92,14 +92,34 @@ def get_grant_type(self) -> InterpolatedString:
     def get_refresh_request_body(self) -> Mapping[str, Any]:
         return self._refresh_request_body.eval(self.config)
 
+    def refresh_access_token(self) -> Tuple[str, Any]:
+        """
+        Returns the refresh token and its lifespan in seconds
+
+        This overrides the parent class method because the parent class assumes the "expires_in" field is always an int representing
+         seconds till token expiry.
+
+        However, this class provides the ability to determine the expiry date of an access token either by using (pseudocode):
+        * expiry_datetime = datetime.now() + seconds_till_access_token_expiry # in this option we have to calculate expiry timestamp, OR
+        * expiry_datetime = parse(response.body["expires_at"]) # in this option the API tells us exactly when access token expires
+
+        :return: a tuple of (access_token, either token_lifespan_in_seconds or datetime_of_token_expiry)
```

Comment:
oof this is a confusing interface

Reply:
yup....

Reply:
FWIW I agree, but this PR just documents the behavior which already exists